### PR TITLE
CVE-2010-4490 and CVE-2010-3247

### PR DESCRIPTION
**Pull request by Ramenous**

### CODE REVIEW

File: cves/CVE-2010-3247.yml [hunk 1]
```diff
@@ -113,8 +116,8 @@ subsystem:
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged. Examples: "clipboard", "gpu", "ssl", "speech", "renderer"
-  answer: Based on the components shown in the bug's blog post.
-  name: Omnibox of the browser's UI
+  answer: This mistake was in the net utilities subsystem where it handled internationalisation and localization
+  name: net
```
Comment (CatherineOsadciw) [inline cves/CVE-2010-3247.yml]: Omnibox might still be relevant for this particular vulnerability (you can list multiple subsystems if applicable)

File: cves/CVE-2010-3247.yml [hunk 2]
```diff
@@ -199,9 +205,15 @@ lessons:
     note:
   complex_inputs:
     applies: true
-    note: The solution used to fix this vulnerability appeared to be by the used of
+    note: |
+      The solution used to fix this vulnerability appeared to be by the used of
```
Comment (CatherineOsadciw) [inline cves/CVE-2010-3247.yml]: I agree complex inputs applies here, but I think it's because of how homographic sequences (complex characters) affected URL rendering.

File: cves/CVE-2010-3247.yml [hunk 3]
```diff
@@ -113,7 +116,8 @@ subsystem:
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged. Examples: "clipboard", "gpu", "ssl", "speech", "renderer"
-  answer: Based on the components shown in the bug's blog post.
+  answer: This mistake was in the net utilities subsystem where it handled internationalisation and localization and based on components in bug's blog post
+  name: net
   name: Omnibox of the browser's UI
```
Comment (CatherineOsadciw) [inline cves/CVE-2010-3247.yml]: Should just be "Omnibox" (long subsystem name values clog up the VHP tag system)

File: cves/CVE-2010-3247.yml [hunk 4]
```diff
@@ -32,9 +32,10 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: Google Chrome doesn't properly respect the characters in urls, allowing
+description: |
+  Google Chrome doesn't properly restrict the characters in urls, allowing
   remote attackers to visually spoof the url bar using sequences of characters that
-  are being used for a different purpose than their intended use.
+  are being used for a different purpose than their intended use (specifically homographic sequences)
```
Comment (bjt6087) [inline cves/CVE-2010-3247.yml]: What are homographic sequences?

Comment (Ramenous) [reply cves/CVE-2010-3247.yml]: In regards to this vulnerability, I'm not 100% sure, but homographic means words that are written in the same form, but have different meanings so I am assuming Japanese characters that are the same/similar with different meanings.

File: cves/CVE-2010-3247.yml [hunk 5]
```diff
@@ -199,9 +206,13 @@ lessons:
     note:
   complex_inputs:
     applies: true
-    note: The solution used to fix this vulnerability appeared to be by the used of
-      regex, which is a specific class for regular expressions, which is a very complex
-      language.
+    note: |
+      The use of homographic sequences which is complex characters affecting the URL
```
Comment (wrs1492) [inline cves/CVE-2010-3247.yml]: Maybe change phrasing to "Complex characters are used in homographic sequences as inputs", or something along those lines.

File: cves/CVE-2010-3247.yml [hunk 6]
```diff
@@ -199,9 +206,13 @@ lessons:
...
+  character_conversion:
+    applies: true
+    note: |
+      The reason why the url spoofing was possible because it did not account and filter for
+      japanese characters in which when typed in a homographic sequence could allow the spoofing
```
Comment (wrs1492) [inline cves/CVE-2010-3247.yml]: You don't need to have both "why the url spoofing was possible", and "could allow the spoofing" in the same sentence. I would recommend removing the second half.

File: cves/CVE-2010-3247.yml [hunk 7]
```diff
@@ -214,9 +225,11 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: I believe the major mistake that lead to the creation of this vulnerability
-    was the lack of testing on visual spoofing of the urls.  If there were more testing
-    of this sort that included the use of foreign characters, such as those that made
-    this an issue, this may have been caught and handled much earlier.  The lack of
+  answer: |
+    The common mistake that led to this vulnerability was the lack of internationalization
```
Comment (wrs1492) [inline cves/CVE-2010-3247.yml]: While this may be unnecessary, I am curious if this specific vulnerability/bug has happened before, or if it is specific to the system. This would be nice to add in this section, as it would mean that Google didn't do their research if it has been a problem in the past.

File: cves/CVE-2010-4490.yml [hunk 8]
```diff
@@ -212,8 +231,10 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: This was a simple coding mistake that had a simple fix.  They did not consider
-    the case of a malformed stream and did not intially check or test for it.  It
-    makes sense that it was overlooked because it is a complex scenario.  In the bug
-    log they were discussing how they were unsure if a malformed stream could easily
-    trigger this, or if it got into a bad state because of some previous root cause.
+  answer: |
+    The mistake they made was a simple coding mistake. They did not include an edge case
```
Comment (wrs1492) [inline cves/CVE-2010-4490.yml]: Is missing an edge case a coding mistake or a testing mistake? I had a similar vulnerability, and mine also involved a strange edge case, and originally I said that it was a coding mistake as well, but other reviewers thought differently.

File: cves/CVE-2010-3247.yml [hunk 9]
```diff
@@ -66,7 +67,7 @@ upvotes_instructions: |
   upvotes to each vulnerability you see. Your peers will tell you how
   interesting they think this vulnerability is, and you'll add that to the
   upvotes score on your branch.
-upvotes: 6
+upvotes: 7
```
Comment (wrs1492) [inline cves/CVE-2010-3247.yml]: Add 3 more upvotes from me.

File: cves/CVE-2010-4490.yml [hunk 10]
```diff
@@ -57,20 +58,21 @@ fixes_vcc_instructions: |
...
 upvotes_instructions: |
-  For the first round, ignore this upvotes number.
+  For the first round, ignore this upvotes number
 
   For the second round of reviewing, you will be giving a certain amount of
   upvotes to each vulnerability you see. Your peers will tell you how
   interesting they think this vulnerability is, and you'll add that to the
   upvotes score on your branch.
-upvotes: 
+upvotes: 1
```
Comment (wrs1492) [inline cves/CVE-2010-4490.yml]: Add 2 upvotes from me.

File: cves/CVE-2010-3247.yml [hunk 11]
```diff
@@ -214,9 +225,11 @@ mistakes:
     Use those questions to inspire your answer. Don't feel obligated to answer
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: I believe the major mistake that lead to the creation of this vulnerability
-    was the lack of testing on visual spoofing of the urls.  If there were more testing
-    of this sort that included the use of foreign characters, such as those that made
-    this an issue, this may have been caught and handled much earlier.  The lack of
+  answer: |
```
Comment (MaeveMonster) [inline cves/CVE-2010-3247.yml]: This is a very verbose, well thought-out answer. Very nice.

File: cves/CVE-2010-4490.yml (hunk 8 above)
Comment (MaeveMonster) [reply cves/CVE-2010-4490.yml]: Yeah I'd say that missing an edge case is a logic error, more so than a coding error.

File: cves/CVE-2010-4490.yml (hunk 10 above)
Comment (MaeveMonster) [reply cves/CVE-2010-4490.yml]: And add 3 from me

File: cves/CVE-2010-3247.yml (hunk 7 above)
Comment (Ramenous) [reply cves/CVE-2010-3247.yml]: I'm not sure if it has happened before, but internalization and localization has always been, I wouldn't say an issue, but rather something that developers tend to overlook or forget when developing applications. It's more of taking consideration of the many possibilities that can be done with foreign characters that are not handled or examined in the code in which this cve is a perfect example of.

File: cves/CVE-2010-4490.yml (hunk 8 above)
Comment (Ramenous) [reply cves/CVE-2010-4490.yml]: Now that I see it, you guys are right. This is more of a logic mistake since they missed out on checking the edge case. Good catch.

File: cves/CVE-2010-3247.yml (hunk 11 above)
Comment (Ramenous) [reply cves/CVE-2010-3247.yml]: Thank you
